pkg/hubble/parser/parser.go

```diff
@@ -75,6 +79,222 @@ func lostEventSourceToProto(source int) pb.LostEventSource {
+		id = n.ID
+		feAddr = &pb.ServiceUpsertNotificationAddr{
+			Ip:   n.Frontend.IP.String(),
+			Port: uint32(n.Frontend.Port),
+		}
+		for _, be := range n.Backends {
+			var ipStr string
+			if be.IP != nil {
+				ipStr = be.IP.String()
+			}
+			beAddrs = append(beAddrs, &pb.ServiceUpsertNotificationAddr{
+				Ip:   ipStr,
+				Port: uint32(be.Port),
+			})
+		}
+		svcType = n.Type
+		svcTrafficPolicy = n.TrafficPolicy
+		svcName = n.Name
+		svcNamespace = n.Namespace
+	}
+	return &pb.AgentEvent{
+		Type: typ,
+		Notification: &pb.AgentEvent_ServiceUpserted{
+			ServiceUpserted: &pb.ServiceUpsertNotification{
+				Id:               id,
+				FrontendAddress:  feAddr,
+				BackendAddresses: beAddrs,
+				Type:             svcType,
+				TrafficPolicy:    svcTrafficPolicy,
+				Name:             svcName,
+				Namespace:        svcNamespace,
+			},
+		},
+	}
+}
+
+func agentNotifyServiceDeletedToProto(typ pb.AgentEventType, msg monitorAPI.AgentNotifyMessage) *pb.AgentEvent {
+	var id uint32
+	if n, ok := msg.Notification.(monitorAPI.ServiceDeleteNotification); ok {
+		id = n.ID
+	}
+	return &pb.AgentEvent{
+		Type: typ,
+		Notification: &pb.AgentEvent_ServiceDeleted{
+			ServiceDeleted: &pb.ServiceDeleteNotification{
+				Id: id,
+			},
+		},
+	}
+}
+
+func agentNotifyMessageToProto(msg monitorAPI.AgentNotifyMessage) *pb.AgentEvent {
+	switch msg.Type {
+	case monitorAPI.AgentNotifyStart:
+		return agentNotifyTimeNotificationToProto(pb.AgentEventType_START, msg)
+	case monitorAPI.AgentNotifyPolicyUpdated:
+		return agentNotifyPolicyNotificationToProto(pb.AgentEventType_POLICY_UPDATED, msg)
+	case monitorAPI.AgentNotifyPolicyDeleted:
+		return agentNotifyPolicyNotificationToProto(pb.AgentEventType_POLICY_DELETED, msg)
+	case monitorAPI.AgentNotifyEndpointRegenerateSuccess:
+		return agentNotifyEndpointRegenNotificationToProto(pb.AgentEventType_ENDPOINT_REGENERATE_SUCCESS, msg)
+	case monitorAPI.AgentNotifyEndpointRegenerateFail:
+		return agentNotifyEndpointRegenNotificationToProto(pb.AgentEventType_ENDPOINT_REGENERATE_FAIL, msg)
+	case monitorAPI.AgentNotifyEndpointCreated:
+		return agentNotifyEndpointUpdateNotificationToProto(pb.AgentEventType_ENDPOINT_CREATED, msg)
+	case monitorAPI.AgentNotifyEndpointDeleted:
+		return agentNotifyEndpointUpdateNotificationToProto(pb.AgentEventType_ENDPOINT_DELETED, msg)
+	case monitorAPI.AgentNotifyIPCacheUpserted:
+		return agentNotifyIPCacheNotificationToProto(pb.AgentEventType_IPCACHE_UPSERTED, msg)
+	case monitorAPI.AgentNotifyIPCacheDeleted:
+		return agentNotifyIPCacheNotificationToProto(pb.AgentEventType_IPCACHE_DELETED, msg)
+	case monitorAPI.AgentNotifyServiceUpserted:
+		return agentNotifyServiceUpsertedToProto(pb.AgentEventType_SERVICE_UPSERTED, msg)
+	case monitorAPI.AgentNotifyServiceDeleted:
+		return agentNotifyServiceDeletedToProto(pb.AgentEventType_SERVICE_DELETED, msg)
+	default:
+		return &pb.AgentEvent{
+			Type: pb.AgentEventType_UNSPECIFIED,
```

Comment:
Not sure how much value this event provides as it. It just tells you that there was a unknown event, but no details or log statement as to what that event was.

I think we should either not forward unknown events at all, or at least provide some more information, e.g. `msg.Type` and maybe even the JSON encoded representation of `msg.Notification`?

Reply:
Agree, will add the suggesteion additional information.